Alignment/OfflineValidation/plugins/EopTreeWriter.cc

```diff
@@ -137,7 +137,7 @@ EopTreeWriter::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup)
    // get geometry
    edm::ESHandle<CaloGeometry> geometry;
    iSetup.get<CaloGeometryRecord>().get(geometry);
-   const CaloGeometry* geo = geometry.product();
+   CaloGeometry* geo = const_cast<CaloGeometry*>(geometry.product());
```

Comment:
my comment https://github.com/cms-sw/cmssw/pull/21440#pullrequestreview-80277065 disappeared for some reason.
The const_cast is still all over the place in this PR and it can not stay, IMHO.

Reply:
Since getFeometry returns shared_ptr cannot use const CaloGeometry* - may be we can cast while accessing each method